### PR TITLE
Search cmakelists only if user asks to configure

### PR DESCRIPTION
**Pull request by vlavati**
### This changes visible behavior while user opens folder/workspace without CMakeLists.txt

The following changes are proposed:

- disable search and select of cmakelists if user asks not to configure project without CMakeLists.txt in root

## The purpose of this change

Comments https://github.com/microsoft/vscode-cmake-tools/issues/3025#issuecomment-1442450001 propose 
to wait to the dialog to select cmakelists and dismiss the prompt.
But it looks that user needs to do that each time, when he opens folder/workspace.

The proposed change allows that user selects his choice to not configure project once (if he persists his choice).

If user wants to configure project later, he is able to run command "CMake: Configure".

### CODE REVIEW

**Comment by vlavati** [comment]
Also affects #3170

File: src/extension.ts [hunk 1]
```diff
@@ -487,31 +487,31 @@ export class ExtensionManager implements vscode.Disposable {
                 shouldConfigure = chosen.doConfigure;
             }
         }
-        if (project) {
-            if (!project.hasCMakeLists()) {
+        if (!project.hasCMakeLists()) {
```
[the remark this answers is not in the document]
Comment (snehara99) [reply src/extension.ts:491]: behavior seems to be as expected with testing

[the remark this answers is not in the document]
Comment (gcampbell-msft) [reply src/extension.ts:491]: It should be okay. There is a check for !project earlier in this method that exists early if it is undefined. 👍
Looks good to me.

[the remark this answers is not in the document]
Comment (vlavati) [reply src/extension.ts:491]: Thanks for review

**Comment by snehara99** [comment]
@vlavati Thanks for your contribution :) As a last step, could you please update [CHANGELOG.md](https://github.com/microsoft/vscode-cmake-tools/blob/main/CHANGELOG.md) in the Improvements section of 1.16 with brief description of your change? You can also (optionally) give yourself credit (see the other entries in the changelog).

**Comment by vlavati** [comment]
@snehara99, It is done.

**Review by snehara99 — APPROVED** [review]
@vlavati This should be merged as soon as the checks finish!

**Pull request merged**